src/ce/crt0.src

```diff
@@ -143,21 +141,67 @@ assert mpTmr2Counter shr 8 = mpTmrCtrl shr 8
 	ld	l, mpTmrCtrl and $FF
 	set	bTmr2Enable, (hl)
 
+	section	.text
+	private	___rerun_handler
+	private	___rerun_handler.callback
+	private	___rerun_handler.data
+	private	___rerun_handler.retval
+___rerun_handler:
+	ld	hl,(_exit.sp)
+	ld	de,0
+.data_size := $ - 3
+	add	hl,de
+	ld	(_exit.sp),hl		; move exit.sp to real ret
+	ld	hl,0
+.retval := $ - 3
+	push	hl
+	ld	hl,0
+.data := $ - 3
+	push	hl
+	call	0
+.callback := $ - 3
+	pop	de, de
+	ret
 
-	section	.start
-	private	start
-start:
+	section .init
+	public	___prgm_run_check
+___prgm_run_check:
```

Comment:
This should be after `save sp for exiting`, so that you can jump straight to the rerun handler and not need all the smc or need to run ans parsing code if main isn't going to be run anyway.  This is not a custom start issue because if it's your job to call main anyway then it's also your job to detect rerun and act accordingly.  However there is still a custom start issue because just putting it in the `.init` section does not guarantee it's location relative to all the other conditional `.init` code.

Reply:
It can't be directly after that because we still need to initialize the bss and constructors.